src/ADAL.PCL/Internal/Telemetry/ApiEvent.cs

```diff
@@ -0,0 +1,155 @@
+        {
+            Tenant = PlatformPlugin.CryptographyHelper.CreateSha256Hash(tenantId);
+            SetEvent(EventConstants.Tenant, Tenant);
+
+            if (userinfo != null)
+            {
+                Idp = userinfo.IdentityProvider;
+                SetEvent(EventConstants.Idp, Idp);
+
+                DisplayableId = PlatformPlugin.CryptographyHelper.CreateSha256Hash(userinfo.DisplayableId);
+                SetEvent(EventConstants.DisplayableId, DisplayableId);
+
+                UniqueId = PlatformPlugin.CryptographyHelper.CreateSha256Hash(userinfo.UniqueId);
+                SetEvent(EventConstants.UniqueId, UniqueId);
+            }
+
+            if (tenantId != null)
+            {
+                Authority = HashTenantIdFromAuthority(authenticator.Authority, tenantId);
+            }
+            else
+            {
+                Authority = authenticator.Authority;
+            }
+            SetEvent(EventConstants.Authority, Authority);
+
+            AuthorityType = authenticator.AuthorityType.ToString();
+            SetEvent(EventConstants.AuthorityType, AuthorityType);
+
+            IsDeprecated = false;
+            SetEvent(EventConstants.IsDeprecated, IsDeprecated.ToString());
+
+            SetEvent(EventConstants.ApiId, apiId);
+        }
+
+        internal string Tenant { get; set; }
+
+        internal string Idp { get; set; }
+
+        internal string Upn { get; set; }
+
+        internal string Authority { get; set; }
+
+        internal string AuthorityType { get; set; }
+
+        internal string ValidateAuthority { get; set; }
+
+        internal string DisplayableId { get; set; }
+
+        internal string IdentityProvider { get; set; }
+
+        internal string UniqueId { get; set; }
+
+        internal bool IsDeprecated { get; set; }
+
+        internal bool ExtendedExpires { get; set; }
+
+        private string HashTenantIdFromAuthority(string authority, string tenantId)
+        {
+            Regex regex = new Regex(tenantId);
+
+            string result = regex.Replace(authority, PlatformPlugin.CryptographyHelper.CreateSha256Hash(tenantId), 5);
+
+            return result;
+        }
+
+        internal override void SetEvent(string eventName, string eventParameter)
+        {
+            if (eventParameter != null)
+            {
+                EventDictitionary.Add(new Tuple<string, string>(eventName, eventParameter));
+            }
+        }
+
+        internal void SetExtraQueryParameters(string extraQueryParameter)
+        {
+            string[] result = extraQueryParameter.Split('&');
+            StringBuilder stringbuilder = new StringBuilder();
+            foreach (string s in result)
+            {
+                if (s.Contains("="))
+                {
+                    stringbuilder.Append(s.Split('=')[0]).Append("&");
+                }
+            }
+            SetEvent(EventConstants.ExtraQueryParameters, stringbuilder.ToString());
```

Comment:
there is an existing query string parsing helper function. In EncodingHelper or available as an extension method.